Next: Draft: Basic readme for nuxt-theme-module (#4701)
DivanteLtd/vue-storefront: packages/core/nuxt-theme-module/README.md

# Nuxt Theme Module
This module:

> - Changes `.nuxt` buildDir to name you set in the config or `.theme` 

Reviewer: it doesn't do this, `.nuxt` and `theme` are completely different things and they're not altering each other

> - Adds `reset.scss` transpiled to CSS to the head (it removes default stylings applied by browser, e.g. blue links)

Reviewer: this should be moved to integration theme tbh, please remove from the doc as we will get rid of that anyway

I already made an issue for that: https://github.com/DivanteLtd/vue-storefront/issues/4702

> - By default, it adds eCommerce routes to your app.

Reviewer: linking the routes would be useful here

- If you set `generate` in the config, it will copy, merge & watch for changes in Agnostic and Integration theme. 

## How to install?
Install module:
```sh
yarn add @vue-storefront/nuxt-theme
```
Add it to build modules in your `nuxt.config.js`:
```js
['@vue-storefront/nuxt-theme'],
```

## Configuration details
If you want to disable autoadding routes you can do it:
```js
['@vue-storefront/nuxt-theme', {
    routes: false
}],
```

> To properly configure `generate` property you have to provide data that will be replaced in EJS templates:

Reviewer: Would be nice to mention that this property applies only to core development as well as an example of such EJS template (just an import)

> # Nuxt Theme Module

Reviewer: ```suggestion
# Nuxt Theme Module

This module is basically our default theme on steroids. It contains additional utilities that make it easier to work with the theme (in projects and as core developers) and allows to disable parts of its features.
```

This module:
- By default, it adds eCommerce [routes](https://github.com/DivanteLtd/vue-storefront/blob/next/packages/core/nuxt-theme-module/routes.js) to your app.

> - If you set `generate` in the config, it will copy, merge & watch for changes in Agnostic and Integration theme. 

Reviewer: ```suggestion
- If you set `generate` in the config, it will copy, merge & watch for changes in agnostic and integration theme. 
```

## How to install
Add dependency:
```sh
yarn add @vue-storefront/nuxt-theme
```
Add it to `buildModules` in your `nuxt.config.js`:
```js
['@vue-storefront/nuxt-theme'],
```

## Configuration details

> If you want to disable autoadding routes you can do it:

Reviewer: ```suggestion
### `routes`
If you want to disable autoadding routes you can do it:
```

```js
['@vue-storefront/nuxt-theme', {
    routes: false
}],
```

> To properly configure `generate` property you have to provide data that will be replaced in EJS templates. This property applies only to the core development:

Reviewer: ```suggestion
### `generate`
To properly configure `generate` property you have to provide data that will be replaced in EJS templates. This property applies only to the core development:
```

```js
['@vue-storefront/nuxt-theme', {
    generate: {
        replace: {
            apiClient: '@vue-storefront/commercetools-api',
            composables: '@vue-storefront/commercetools'
        }
    }
}],
```
Example EJS template where it will be pasted:
```js
import { useCategory } from '<%= options.generate.replace.composables %>';
```

> Changing target directory where Agnostic and Integration themes will be merged and copied in Core Development Mode (default: `.theme`):

Reviewer: ```suggestion
If you want to change the target directory where agnostic and integration themes will be merged and copied during core development change the `generate.path` property (default: `.theme`):
```
